src/Client.php

```diff
@@ -58,13 +64,15 @@ final class Client implements FlushableClientInterface
      * @param Options               $options      The client configuration
      * @param TransportInterface    $transport    The transport
      * @param EventFactoryInterface $eventFactory The factory for events
+     * @param LoggerInterface|null  $logger       The PSR-3 logger
      */
-    public function __construct(Options $options, TransportInterface $transport, EventFactoryInterface $eventFactory)
+    public function __construct(Options $options, TransportInterface $transport, EventFactoryInterface $eventFactory, ?LoggerInterface $logger = null)
     {
         $this->options = $options;
         $this->transport = $transport;
         $this->eventFactory = $eventFactory;
         $this->integrations = Handler::setupIntegrations($options->getIntegrations());
+        $this->logger = $logger;
```

Comment:
Using the `NullLogger` as a `null` fallback would reduce null checks a lot in our code.

```suggestion
        $this->logger = $logger ?? new NullLogger();
```

Reply:
Done. A downside that I see however is that we're instantiating the logger in multiple places, which I don't really like, and also it's not really explicit of course that the logger is doing nothing